### PR TITLE
Added Arithmetic Equations Visualization, New test cases added (Value…

### PR DESCRIPTION
**Pull request by wintersoil**
…s.elm), partial AST generation done (AbstractTreeVisualizer.elm) for Arithmetic Operations

# Terms

> THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE TERMS OF THE CCLA DATED 2017-11-07 WITH FINOS/LINUX FOUNDATION (FORMERLY THE SYMPHONY SOFTWARE FOUNDATION CCLA).
>
> THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

File: redistributable/SpringBoot/generated.iml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AttilaMihaly) [inline redistributable/SpringBoot/generated.iml:1]: Can you remove this file from your PR? I don't think this was intentional.

File: src/Morphir/Visual/Components/AritmeticExpressions.elm [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    case ao of
+        Add ->
+            "Add"
+
+        Subtract ->
+            "Subtract"
+
+        Multiply ->
+            "Multiply"
+
```
Comment (AttilaMihaly) [inline src/Morphir/Visual/Components/AritmeticExpressions.elm:101]: Is this used anywhere?

File: src/Morphir/Visual/Components/AbstractTreeVisualizer.elm [hunk 3]
```diff
@@ -0,0 +1,272 @@
+module Morphir.Visual.Components.AbstractTreeVisualizer exposing (..)
```
Comment (AttilaMihaly) [inline src/Morphir/Visual/Components/AbstractTreeVisualizer.elm:1]: Is this just a sample app? If so, can you move it to the `tests` folder or within the `cli` folder?

File: src/Morphir/Visual/ViewValue.elm [hunk 4]
```diff
@@ -96,6 +102,15 @@ viewValueByValueType ctx argumentValues typedValue =
...
         ViewBoolOperatorTree.view (viewValueByLanguageFeature ctx argumentValues) boolOperatorTree
 
+    else if Basics.isNumber valueType then
+        let
+            arithmeticOperatorTree : ArithmeticOperatorTree
+            arithmeticOperatorTree =
+                ArithmeticOperatorTree.fromArithmeticTypedValue typedValue
+        in
+        ViewArithmetic.view (viewValueByLanguageFeature ctx argumentValues) arithmeticOperatorTree
+        -- Element.column [] [ ViewArithmetic.view (viewValueByLanguageFeature ctx argumentValues) arithmeticOperatorTree, Element.html (div [ Html.Attributes.style "display" "none", Html.Attributes.style "padding" "10", Html.Attributes.style "background-color" "rgb (50 50 50)", Html.Attributes.style "font-size" "16", Html.Attributes.style "color" "rgb(255 78 185 255)" ] [ outerView [] (parseArithmeticOperatorTree arithmeticOperatorTree) ]) ]
```
Comment (AttilaMihaly) [inline src/Morphir/Visual/ViewValue.elm:112]: Can you remove these comments.

File: src/Morphir/Visual/ViewValue.elm [hunk 5]
```diff
@@ -176,3 +194,62 @@ viewValueByLanguageFeature ctx argumentValues value =
...
+            "AVL (Variable) -> " ++ Name.toTitleCase name
+
+        Reference va fQName ->
+            "AVL (Reference) -> " ++ FQName.toString fQName
+
+        Apply va value2 value3 ->
+            "AVL (Apply): " ++ helperFunctionValue value2 ++ " __ " ++ helperFunctionValue value3
+
+        _ ->
+            "Some other format"
```
Comment (AttilaMihaly) [inline src/Morphir/Visual/ViewValue.elm:255]: Do we need all this here?

File: tests-integration/reference-model/src/Morphir/Reference/Model/Values.elm [hunk 6]
```diff
@@ -4,6 +4,81 @@ import Morphir.Reference.Model.Types as Types exposing (Custom(..), FooBarBazRec
...
+unemploymentRate : Float -> Float -> Float
+unemploymentRate unemployed employed =
+    unemployed / employed
+
+
+amortization : Float -> Float -> Float -> Float -> Float
+amortization p r n t =
+    (p * r / n) / (1 - (1 + r / n) - n * t)
+
+
```
Comment (AttilaMihaly) [inline tests-integration/reference-model/src/Morphir/Reference/Model/Values.elm:81]: Can you move these to a separate file? Say `Formulas.elm` or something similar.

**Review by AttilaMihaly — CHANGES_REQUESTED** [review]

File: redistributable/SpringBoot/generated.iml (hunk 1 above)
Comment (wintersoil) [reply redistributable/SpringBoot/generated.iml]: removed

File: src/Morphir/Visual/Components/AritmeticExpressions.elm (hunk 2 above)
Comment (wintersoil) [reply src/Morphir/Visual/Components/AritmeticExpressions.elm]: No, removed

File: src/Morphir/Visual/ViewValue.elm (hunk 4 above)
Comment (wintersoil) [reply src/Morphir/Visual/ViewValue.elm]: Moved comment to AbstractTreeVisualizer.elm,
can be added back here to get visual of Arithmetic AST (debugging)

File: src/Morphir/Visual/ViewValue.elm (hunk 5 above)
Comment (wintersoil) [reply src/Morphir/Visual/ViewValue.elm]: moved to to AbstractTreeVisualizer.elm in tests/Morphir

File: tests-integration/reference-model/src/Morphir/Reference/Model/Values.elm (hunk 6 above)
Comment (wintersoil) [reply tests-integration/reference-model/src/Morphir/Reference/Model/Values.elm]: Moved, done

**Review by AttilaMihaly — APPROVED** [review]

**Pull request merged**